### PR TITLE
docs update

### PR DESCRIPTION
**Pull request by yangcao77**
Signed-off-by: Stephanie <yangcao@redhat.com>

**Please specify the area for this PR**
registry-library doc
registry server doc

**What does does this PR do / why we need it**:
This PR updates the registry library doc with v2index changes. 
also move the registry server REST API change to this repo, the landing page should point to this link afterwards

**Which issue(s) this PR fixes**:

https://github.com/devfile/api/issues/838

**PR acceptance criteria**:

- [ ] Test (WIP) 

- [x] Documentation (WIP)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by yangcao77** [comment]
@michael-valdron  can you help adding the starter project changes to those two docs as well? Thanks

**Comment by michael-valdron** [comment]
> @michael-valdron can you help add the starter project changes to those two docs as well? Thanks

Done!

**Comment by yangcao77** [comment]
@kim-tsao Micheal did not update the registry library, so no need to update that doc. but perhaps an API should be introduced to the library for starterprojects @michael-valdron @johnmcollier Any thoughts?

**Comment by michael-valdron** [comment]
> > > @michael-valdron can you help add the starter project changes to those two docs as well? Thanks
> > 
> > 
> > Done!
> 
> @michael-valdron can you also update this doc? https://github.com/devfile/registry-support/blob/main/registry-library/README.md It's the documentation for the registry library API

For Starter Projects I did not update the [registry-library](https://github.com/devfile/registry-support/tree/main/registry-library), @yangcao77 and I only updated the following:

- [devfile/library](https://github.com/devfile/library) - filters, bug fix
- [index/generator](https://github.com/devfile/registry-support/tree/main/index/generator) - util functions
- [index/server](https://github.com/devfile/registry-support/tree/main/index/server) - endpoints

**Comment by yangcao77** [comment]
@michael-valdron  since registry library is just providing library API to interact with registry server, it should reflect all features that REST APIs provide. I do think we need to include the starterproject APIs in registry library.  
@kim-tsao @johnmcollier @elsony what do you guys think?

**Comment by johnmcollier** [comment]
I won't pile on to the plus 1s 🙂 , but can we get an issue open then for the registry library changes?

File: index/server/registry-REST-API.adoc [hunk 1]
```diff
@@ -0,0 +1,2088 @@
...
+|Registry stack name
+
+|Version
+|Specific version of the stack, or `latest`
+
+|===
+
+=== Request body
+The request body must be empty.
+=
```
Comment (kim-tsao) [inline index/server/registry-REST-API.adoc:1927]: Remove, `Request Example` subheading is not formatted properly

**Review by kim-tsao — COMMENTED** [review]
In general, the doc is pretty good but should we consider truncating some of the lengthier responses?  Should we also have a table of contents so that users can directly navigate to the API they want?

File: index/server/registry-REST-API.adoc [hunk 2]
```diff
@@ -0,0 +1,2088 @@
...
+|The URL/ingress that exposes registry service
+
+|Stack
+|Registry stack name
+
+|===
+
+=== Request body
+The request body must be empty.
+=
```
[the remark this answers is not in the document]
Comment (yangcao77) [reply index/server/registry-REST-API.adoc]: updated

File: index/server/registry-REST-API.adoc (hunk 1 above)
Comment (yangcao77) [reply index/server/registry-REST-API.adoc]: updated

**Comment by yangcao77** [comment]
> In general, the doc is pretty good but should we consider truncating some of the lengthier responses? Should we also have a table of contents so that users can directly navigate to the API they want?

Good point. I added a navigation table at top of the doc. Can you review it as well?

**Comment by kim-tsao** [comment]
> > In general, the doc is pretty good but should we consider truncating some of the lengthier responses? Should we also have a table of contents so that users can directly navigate to the API they want?
> 
> Good point. I added a navigation table at top of the doc. Can you review it as well?

yes, I'll review it

**Review by kim-tsao — APPROVED** [review]

**Comment by michael-valdron** [comment]
> I won't pile on to the plus 1s slightly_smiling_face , but can we get an issue open for the registry library changes?

devfile/api#842

@johnmcollier @yangcao77 @kim-tsao @elsony

**Pull request merged**